kola/tests/locksmith/locksmith.go

```diff
@@ -233,32 +233,40 @@ func locksmithCluster(c cluster.TestCluster) {
 
 	c.MustSSH(machs[0], "locksmithctl status")
 
-	ctx := context.Background()
-	wg := worker.NewWorkerGroup(ctx, len(machs))
```

Comment:
Can the diff just be 

```
ctx, cleanup := context.WithTimeout(context.TODO(), 10 * time.Minute)
defer cleanup()
```

and not make any other changes?

I don't think the util timeout function is useful if we already are using contexts since they have `WithTimeout` already

Reply:
Just using `context.WithTimeout` doesn't cause the test to fail when a timeout occurs. I'll look into ways of making it properly bubble out the failure.

Reply:
I think the failure to bubble up the failure is still in the locksmith code after reading more.

Basically, I think it's expected the function passed to `wg.Start` should bubble up context errors if they're relevant.

In this case, they are, so I think the right fix is to have the worker func defined a few lines below return `ctx.Err()` after the machine check.

It does seem a little awkward that the timeout will fail the waitgroup if it happens before all work is scheduled, but requires the work to bubble it up if it happens later, but I think it's overall reasonable and in-line with how context is meant to be used.

Reply:
Actually, after looking a bit more, it seems like `platform.CheckMachine` should probably take a `ctx` as an argument and respect its timeout since that's where the time's really being spent: retrying there.

Reply:
Pushed a change that adds a `context.Context` variable to `platform.CheckMachine`.